### ISSUE TITLE
Generated Activity_ does not override onActivityResult for WRITE_SETTINGS permission

### ISSUE THREAD

**Issue by kaiyan910** [AleksanderMielczarek/AndroidAnnotationsPermissionsDispatcherPlugin#9]
HI, I am using latest version 2.0.1, 
and one of my activity require the permission WRITE_SETTINGS, 
the PermissionsDispatcher generated correctly with the method "onActivityResult". 

```
static void onActivityResult(xxxActivity target, int requestCode) {
    switch (requestCode) {
      case REQUEST_WRITESETTINGS:
      if (PermissionUtils.hasSelfPermissions(target, PERMISSION_WRITESETTINGS) || Settings.System.canWrite(target)) {
        target.writeSettings();
      } else {
        target.showDeniedForWriteSettings();
      }
      break;
      default:
      break;
    }
  }
```

However the generated Activity_ override onRequestPermissionsResult instead of onActivityResult

```
@Override
public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
 super.onRequestPermissionsResult(requestCode, permissions, grantResults);
 xxxActivityPermissionsDispatcher.onRequestPermissionsResult(this, requestCode, grantResults);
 permissionDispatcherCalled_ = false;
}
```
And because the PermissionsDispatcher does not have the method onRequestPermissionsResult, so it is marked as error.

**Comment by AleksanderMielczarek** [comment]
Hi,

I don't understand why you mix `onActivityResult` with `onRequestPermissionsResult`. These are two different callbacks. Please attach full activity code.

Best regards
Aleksander Mielczarek

**Comment by kaiyan910** [comment]
I am not mixing 2 together, requesting WRITE_SETTINGS not like other permission, you need to set it on the settings page, thats why the generate class implement onActivityResult but not onRequestPermissionsResult.

however using the 2.0 version the generated activity class just override the onRequestPermissionsResult and try to call onRequestPermissionsResult from the xxxActivityPermissionsDispatcher which does not exist. 

You may try to request WRITE_SETTINGS for more details the code above is the generated code

**Comment by AleksanderMielczarek** [comment]
Hi,

I understand now. Thank you for provided example. 

In that case plugin needs to check what permissions are requested and generate code in `onActivityResult` and `onRequestPermissionsResult` respectively. I will check and try to fix this one.

**Comment by kaiyan910** [comment]
My english is bad XD sorry about that

Thanks for your help!

**Comment by AleksanderMielczarek** [comment]
@kaiyan910 fixed in 2.0.2. Please try new version 😃
